src/webpack.config.js

```diff
@@ -32,44 +32,101 @@ module.exports = function (args) {
 		}));
 	}
 
-	return {
-		entry: {
-			'src/main': [
-				path.join(basePath, 'src/main.styl'),
-				path.join(basePath, 'src/main.ts')
-			]
-		},
-		devtool: 'source-map',
-		resolve: {
-			root: [ basePath ],
-			extensions: ['', '.ts', '.tsx', '.js'],
-			alias: {
-				rxjs: '@reactivex/rxjs/dist/amd'
-			}
-		},
-		resolveLoader: {
-			root: [ path.join(__dirname, 'node_modules') ]
-		},
-		module: {
-			preLoaders: [
-				{
-					test: /dojo-.*\.js$/,
-					loader: 'source-map-loader'
-				}
-			],
-			loaders: [
-				{ test: /src[\\\/].*\.ts?$/, loader: 'umd-compat-loader!ts-loader' },
-				{ test: /\.js?$/, loader: 'umd-compat-loader' },
-				{ test: /\.html$/, loader: 'html' },
-				{ test: /\.(jpe|jpg|png|woff|woff2|eot|ttf|svg)(\?.*$|$)/, loader: 'file?name=[path][name].[hash:6].[ext]' },
-				{ test: /\.styl$/, loader: ExtractTextPlugin.extract(['css-loader?sourceMap', 'stylus-loader']) },
-				{ test: /\.css$/, loader: 'style-loader!css-loader?modules' },
-			]
-		},
-		plugins: plugins,
-		output: {
-			path: path.resolve('./dist'),
-			filename: '[name].js'
-		}
-	};
+	const webpackConfigs = [{
+        entry: {
+            'src/main': [
+                path.join(basePath, 'src/main.styl'),
+                path.join(basePath, 'src/main.ts')
+            ]
+        },
+        devtool: 'source-map',
+        resolve: {
+            root: [ basePath ],
+            extensions: ['', '.ts', '.tsx', '.js'],
+            alias: {
+                rxjs: '@reactivex/rxjs/dist/amd'
+            }
+        },
+        resolveLoader: {
+            root: [ path.join(__dirname, 'node_modules') ]
+        },
+        module: {
+            preLoaders: [
+                {
+                    test: /dojo-.*\.js$/,
+                    loader: 'source-map-loader'
+                }
+            ],
+            loaders: [
+                { test: /src[\\\/].*\.ts?$/, loader: 'umd-compat-loader!ts-loader' },
+                { test: /\.js?$/, loader: 'umd-compat-loader' },
+                { test: /\.html$/, loader: 'html' },
+                { test: /\.(jpe|jpg|png|woff|woff2|eot|ttf|svg)(\?.*$|$)/, loader: 'file?name=[path][name].[hash:6].[ext]' },
+                { test: /\.styl$/, loader: ExtractTextPlugin.extract(['css-loader?sourceMap', 'stylus-loader']) },
+                { test: /\.css$/, loader: 'style-loader!css-loader?modules' },
+            ]
+        },
+        plugins: plugins,
+        output: {
+            path: path.resolve('./dist'),
+            filename: '[name].js'
+        }
+    }];
+	if (args.withTests) {
```

Comment:
would it not be better to mix the below over the config rather then duplicating?